### PR TITLE
use json flags

### PR DESCRIPTION
**Pull request by prolic**
ping @enumag @basz

### CODE REVIEW

File: src/StreamMetadata.php [hunk 1]
```diff
@@ -16,6 +16,7 @@
 use JsonSerializable;
 use Prooph\EventStoreClient\Common\SystemMetadata;
 use Prooph\EventStoreClient\Exception\InvalidArgumentException;
+use Prooph\EventStoreClient\Exception\JsonException;
 
 class StreamMetadata implements JsonSerializable
```
[the remark this answers is not in the document]
Comment (prolic) [reply src/StreamMetadata.php]: Ah damn it, my bad :)

**Comment by prolic** [comment]
A custom JsonSerializer / JsonDeserializer is really not needed. The only places where json_encode is used is in `setStreamMetadataAsync` and `setSystemSettingsAsync`. For the first option you can use `setRawStreamMetadataAsync` instead and for the latter append to the settings stream manually yourself.

The only other usage is the UserClient, where I this this is really not needed. You control your event data 100%, see `\Prooph\EventStoreClient\EventData` class.

**Comment by enumag** [comment]
StreamMetadata now looks asymetrical with jsonSerialize that returns an array and jsonUnserialize that actually accepts json string. Maybe replace jsonUnserialize with createFromArray and move json out of that class?

**Comment by prolic** [comment]
> StreamMetadata now looks asymetrical with jsonSerialize that returns an array and jsonUnserialize that actually accepts json string. Maybe replace jsonUnserialize with createFromArray and move json out of that class?

Yes, good point

**Pull request merged**